app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,57 @@
+class  ReviewsController < ApplicationController
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @buddy = Buddy.find(params[:buddy_id])
+    @review = Review.new(review_params)
+    @review.buddy_id = @buddy.id
+    @review.user_id = current_user.id
+    if @review.save
+      flash[:notice] = "You've successfully submitted a review!"
+      redirect_to buddy_path(@buddy)
+    else
+      flash[:alert] = @review.errors.full_messages
+      redirect_to buddy_path(@buddy)
+    end
+  end
+
+  def edit
+    @review = current_user.reviews.find(params[:id])
+    @buddy = Buddy.find(params[:buddy_id])
+  end
+
+  def update
+    @review = current_user.reviews.find(params[:id])
+    @buddy = Buddy.find(params[:buddy_id])
+    if current_user.id == @review.user_id
```

Comment:
This check is redundant since you are only pulling from reviews that the current user owns:

```ruby
@review = current_user.reviews.find(params[:id])
```